lebedigital/openbis/expstep.py

```diff
@@ -0,0 +1,531 @@
+import logging
+import os
+import shutil
+from math import isnan
+from sys import exit
+
+import pandas as pd
+import yaml
+from termcolor import colored
+
+from lebedigital.openbis.interbis import Interbis
+
+
+class ExpStep:
+    def __init__(
+            self,
+            name: str = '',
+            type: str = '',
+            metadata: dict = None,
+            space: str = "",
+            project: str = "",
+            collection: str = "",
+            parents: list = None,
+            identifier: str = '',
+            permId: str = '',
+            sample_object = None,
+            datasets: list = None,
+            dataset_codes: list = None,
+    ) -> None:
+        """Creates an ExpStep Object as general python interface to the Experimental Step OpenBis object
+
+        Args:
+            name (str, optional): name of the experiment. Defaults to ''.
+            type (str, optional): type of the experiment, has to be a defined sample type in the Datastore. Defaults to ''.
+            metadata (dict, optional): metadata of your data, is defined by the sample type. Defaults to None.
+            space (str, optional): space in which the experiment should be saved. Defaults to "".
+            project (str, optional): project in which the experiment should be saved. Defaults to "".
+            collection (str, optional): collection in which the experiment should be saved. Defaults to "".
+            parents (list, optional): parents of the experiment. Defaults to None.
+            identifier (str, optional): verbose identifier of the sample, indicates the space and project. Defaults to ''.
+            permId (str, optional): PermID identifier of the sample. Defaults to ''.
+            sample_object (Pybis, optional): the pybis sample object of the sample. Defaults to None.
+            datasets (list, optional): list of dataset objects saved under the experimental step. Defaults to None.
+            dataset_codes (list, optional): the PermIDs of the datasets. Defaults to None.
```

Comment:
what is a dataset_code? Could you add a discussion on all those terms into a the documentation?

Reply:
I could write a readme about it or I could add a page explaining it when I'll work on the docs for the code.